src/Compatibility/Microsoft.DotNet.PackageValidation/ValidatePackage.cs

```diff
@@ -55,5 +78,20 @@ protected override void ExecuteCore()
                 logger.GenerateSuppressionsFile(CompatibilitySuppressionFilePath);
             }
         }
+
+        private Assembly ResolverForRoslyn(object sender, ResolveEventArgs args)
+        {
+            AssemblyName name = new(args.Name);
+            if (name.Name == "Microsoft.CodeAnalysis" || name.Name == "Microsoft.CodeAnalysis.CSharp")
+            {
+                Assembly asm = Assembly.LoadFrom(Path.Combine(RoslynAssemblyPath, $"{name.Name}.dll"));
+                if (asm.GetName().Version < name.Version)
+                {
+                    throw new Exception(Resources.UpdateSdkVersion);
```

Comment:
Might be helpful to list more details about what assembly was being loaded, what version was expected and what version was found?  Also, the action taken would be to use a newer compiler.  In the common case this assembly would come with the SDK and the compiler version in the SDK would always be sufficient.  Users would hit this error if they tried to use the OOB compilers, or tried to consume the OOB PackageValidation.

Reply:
It will also throw a load error with information regarding the assembly  its trying to load.

Reply:
That won't list why we rejected it though.  Might be helpful to include at least the version info.